TA_dataset/bin/dataset_api.py

```diff
@@ -25,6 +24,11 @@
 )
 
 
+class APIException(Exception):
+    def __init__(self, payload):
+        self.payload = payload
```

Comment:
Would it also make sense to store other information such as the response status code?

Reply:
Please also add a comment on what `payload` contains / stores - API response body.

Reply:
Currently it's passed to `search_error_exit` which does not care. I am not sure, whether it's useful for user to know, that it was 401 or 503. I hope that this is captured in that message.